src/pytest_plugins/test_filler/test_filler.py

```diff
@@ -468,6 +468,29 @@ def pytest_generate_tests(metafunc):
             )
 
 
+@pytest.hookimpl(trylast=True)
+def pytest_collection_modifyitems(config, items):
+    """
+    Remove pre-Merge tests parametrized to generate hive type fixtures; these
+    can't be used in the Hive Pyspec Simulator.
+
+    This can't be handled in this plugins pytest_generate_tests() as the fork
+    parametrization occurs in the forks plugin.
+    """
+    for item in items[:]:  # use a copy of the list, as we'll be modifying it
+        if isinstance(item, EIPSpecTestItem):
+            continue
+        if not is_fork(item.callspec.params["fork"], Merge):
+            if ("state_test" in item.callspec.params) and item.callspec.params[
```

Comment:
`state_test` is fine pre-merge, is just the hive that's incompatible.

Reply:
This only removes `state_test_hive` formats, not all of the `state_test` formats, L486.

But I don't think any of `state_test_hive` should ever be present, as neither `StateTest.fixture_formats()`, or `BlockchainTest.fixture_formats() return it.

Perhaps we should remove it entirely from the `FixtureFormats` enum? Or was there another intent to use this format?

Reply:
I totally missed the name check! Thanks!

I agree there is no use for it at the moment, I will remove it after merging this PR :+1: